runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java

```diff
@@ -1591,15 +1607,23 @@ public void testTemplateRunnerLoggedErrorForFile() throws Exception {
   }
 
   @Test
-  public void testWorkerHarnessContainerImage() {
+  public void testGetContainerImageForJobFromOption() {
     DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
 
-    // default image set
-    options.setWorkerHarnessContainerImage("some-container");
+    // When image option is set, should use that exact image.
+    options.setSdkContainerImage("some-container");
     assertThat(getContainerImageForJob(options), equalTo("some-container"));
 
+    // When image option is empty, container image should not return default
+    // to allow for use of pipeline worker jar.
+    options.setSdkContainerImage("");
```

Comment:
This seems load-bearing enough to require at least a unit test.

Reply:
Agree. Empty string makes me nervous. How about a string like `"this is a test"` that you set to a variable early in the test case so anyone reading understands that they are defined to be the same.

Reply:
made a list of test cases with a comment about testing empty string specifically